[Enhancement]: unstable ut regarding channel rebalance

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

FAIL: TestConsistentHashRegisterPolicy (0.00s)
2023-11-13T04:02:37.6116973Z     --- PASS: TestConsistentHashRegisterPolicy/first_register (0.00s)
2023-11-13T04:02:37.6118179Z     --- FAIL: TestConsistentHashRegisterPolicy/rebalance_after_register (0.00s)

### Why is this needed?

unstable ut affect the result of ci

### Anything else?

[logs_196685.zip](https://github.com/milvus-io/milvus/files/13331314/logs_196685.zip)

--- a/internal/datacoord/policy_test.go
+++ b/internal/datacoord/policy_test.go
@@ -112,11 +112,18 @@ func TestConsistentHashRegisterPolicy(t *testing.T) {
 		assert.NotNil(t, updates)
 		assert.Equal(t, 1, len(updates))
 		// No Delete operation will be generated
-		assert.EqualValues(t, &ChannelOp{Type: Add, NodeID: 1, Channels: []RWChannel{channels[0]}}, updates[0])
-		assert.ElementsMatch(t,
-			NewChannelOpSet(NewAddOp(1, channels[0])).Collect(),
-			updates,
-		)
+
+		assert.Equal(t, 1, len(updates[0].GetChannelNames()))
+		channel := updates[0].GetChannelNames()[0]
+
+		// Not stable whether to balance chan-1 or chan-2
+		if channel == "chan-1" {
+			assert.EqualValues(t, &ChannelOp{Type: Add, NodeID: 1, Channels: []RWChannel{channels[0]}}, updates[0])
+		}
+
+		if channel == "chan-2" {
+			assert.EqualValues(t, &ChannelOp{Type: Add, NodeID: 1, Channels: []RWChannel{channels[1]}}, updates[0])
+		}
 	})
 }
 

